soc/arm/nordic_nrf/timing.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arch/arm/aarch32/arch.h>
+#include <kernel.h>
+#include <sys_clock.h>
+#include <timing/timing.h>
+#include <nrfx.h>
+
+#if defined(CONFIG_NRF_RTC_TIMER)
+
+
+#define CYCLES_PER_SEC (16000000 / (1 << NRF_TIMER2->PRESCALER))
```

Comment:
I think it's good to have a build-time assert here, since it is possible in the future to have nRF SoCs without a 3rd timer instance (TIMER2)

Reply:
right, that why the ifdef above, but for other time sources we will have other implementation of those functions.

Reply:
> > @nashif @dcpleung if you have this board could you investigate?
> 
> This issue existed in master already with nrf51 with latency_measure and other test using the original functions on top of RTC timer. I am not sure how I can fix those, so if you a suggestion, please go ahead, but I do not think they should block this PR. We could disable timeing api for this SoC until we have an appropriate implementation that works.

I don't want to block this PR, but you should at least address the second point i made above (i.e. how to calculate the averages).